src/hooks/useCompany.ts

```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react';
+
+const useCompany = () => {
+  const [company, setCompany] = useState<string>('');
+
+  const handleCompanySelect = (value: string) => {
+    setCompany(value);
+  };
+
+  return { company, handleCompanySelect };
+};
```

Comment:
```suggestion
const useCompany = () => {
  const [company, setCompany] = useState<CardCompany | null>(null);

  const handleCompanySelect = (value: CardCompany) => {
    setCompany(value);
  };

  return { company, handleCompanySelect };
};
```
공용 훅이 아닌 도메인 관련 훅인 것 같아서 타입을 좁혀봐도 좋을 것 같슴다~